### PR TITLE
Rename `source` field in Filebeat 

### PR DESCRIPTION
**Pull request by ruflin**
The source field in Filebeat was used to store the file path for logs or the source ip for syslog, udp, tcp input. As source is in ECS an object the fields are now moved to ECS pattern.

* For UDP, TCP, syslog input the source field is converted to log.source.ip
* For the log input the source field is converted to log.file.path

Done:

* Test files updated
* Changelog updated
* Migration file updated
* `source` removed from fields.yml, two new fields added

### CODE REVIEW

File: filebeat/input/udp/input.go [hunk 1]
```diff
@@ -74,7 +74,9 @@ func NewInput(
 			},
 			Fields: common.MapStr{
 				"message": string(data),
-				"source":  metadata.RemoteAddr.String(),
+				"source": common.MapStr{
```
Comment (ruflin) [inline filebeat/input/udp/input.go]: @ph Does this make sense for udp, tcp and syslog?

**Comment by ruflin** [comment]
@ph @andrewkroh @webmat I actually have second thoughts here about the proposed fields. `file.path` and `source.ip` can both show up in the log files itself. What if the apache logs are sent through syslog and both have `source.ip`.

The `file.path` and `source.ip` are meta information about a log event. Because of this I propose `log.file.path` and `log.source.ip`. Both are not ECS fields but follow the naming patterns.

**Comment by webmat** [comment]
Since this is one field being split in two locations (file path vs IP), we can't list it in `dev-tools/ecs-migration.yml`, correct?

**Comment by ruflin** [comment]
@webmat I'll figure out a way to list it

File: CHANGELOG.asciidoc [hunk 2]
```diff
@@ -20,6 +20,7 @@ https://github.com/elastic/beats/compare/v6.4.0...master[Check the HEAD diff]
 - Use `initial_scan` action for new paths. {pull}7954[7954]
 
 *Filebeat*
+- Move source to log.file.path and log.source.ip
```
Comment (andrewkroh) [inline CHANGELOG.asciidoc]: I'd use "Rename" rather than "Move".

File: filebeat/_meta/fields.common.yml [hunk 3]
```diff
@@ -3,13 +3,20 @@
...
+      required: false
       description: >
         The file from which the line was read. This field contains the absolute path to the file.
         For example: `/var/log/system.log`.
 
+    - name: log.source.ip
+      type: ip
+      required: false
+      description: >
+        Source ip from which the log event was read / sent from.
```
Comment (andrewkroh) [inline filebeat/_meta/fields.common.yml]: `s/ip/IP/`

File: CHANGELOG.asciidoc (hunk 2 above)
Comment (ruflin) [reply CHANGELOG.asciidoc]: fixed

File: filebeat/_meta/fields.common.yml (hunk 3 above)
Comment (ruflin) [reply filebeat/_meta/fields.common.yml]: fixed

File: CHANGELOG.asciidoc [hunk 4]
```diff
@@ -20,7 +20,7 @@ https://github.com/elastic/beats/compare/v6.4.0...master[Check the HEAD diff]
 - Use `initial_scan` action for new paths. {pull}7954[7954]
 
 *Filebeat*
-
```
Comment (adriansr) [inline CHANGELOG.asciidoc]: Nit: You've removed the empty line between the header (Filebeat) and the list of changes

File: x-pack/filebeat/module/suricata/eve/test/eve-alerts.log-expected.json [hunk 5]
```diff
@@ -16,6 +16,7 @@
         "http.request.method": "GET", 
         "http.response.status_code": "200", 
         "input.type": "log", 
+        "log.file.path": "/Users/ruflin/Dev/gopath/src/github.com/elastic/beats/x-pack/filebeat/module/suricata/eve/test/eve-alerts.log", 
         "log.offset": 0,
```
Comment (adriansr) [inline x-pack/filebeat/module/suricata/eve/test/eve-alerts.log-expected.json]: As this field is ignored by test_modules.py, can we get rid of this entries?

File: CHANGELOG.asciidoc (hunk 4 above)
Comment (ruflin) [reply CHANGELOG.asciidoc]: Will fix this in a follow up PR

File: x-pack/filebeat/module/suricata/eve/test/eve-alerts.log-expected.json (hunk 5 above)
Comment (ruflin) [reply x-pack/filebeat/module/suricata/eve/test/eve-alerts.log-expected.json]: Good question. Will need to investigate. Will also tackle this in one of the other PR's that is conflicting with this one as soon as it's merged if that is ok with you?

Comment (ruflin) [reply x-pack/filebeat/module/suricata/eve/test/eve-alerts.log-expected.json]: You got lucky, have to rebase now anyways. Will fix these issues.

**Comment by ruflin** [comment]
Ready for an other round of reviews. Failing tests should not be related.

**Pull request merged**